pallets/corporate-actions/src/ballot.rs

```diff
@@ -0,0 +1,670 @@
+        ///
+        /// # Errors
+        /// - `UnauthorizedAsAgent` if `origin` is not `ticker`'s sole CAA (owner is not necessarily the CAA).
+        /// - `NoSuchBallot` if `ca_id` does not identify a ballot.
+        /// - `VotingAlreadyStarted` if `start >= now`, where `now` is the current time.
+        /// - `NumberOfChoicesOverflow` if the total choice in `meta` overflows `usize`.
+        #[weight = 950_000_000]
+        pub fn change_meta(origin, ca_id: CAId, meta: BallotMeta) {
+            // Ensure origin is CAA, a ballot exists, start is in the future.
+            let caa = <CA<T>>::ensure_ca_agent(origin, ca_id.ticker)?;
+            Self::ensure_ballot_not_started(Self::ensure_ballot_exists(ca_id)?)?;
+
+            // Compute number-of-choices-in-motion cache.
+            let choices = Self::derive_motion_num_choices(&meta.motions)?;
+
+            // Commit metadata to storage + emit event.
+            MotionNumChoices::insert(ca_id, choices);
+            Metas::insert(ca_id, meta.clone());
+            Self::deposit_event(Event::<T>::MetaChanged(caa, ca_id, meta));
+        }
+
+        /// Amend RCV support for the ballot of the CA identified by `ca_id`.
+        ///
+        /// ## Arguments
+        /// - `origin` which must be a signer for the CAA of `ca_id`.
+        /// - `ca_id` identifies the attached ballot's CA.
+        /// - `rcv` specifies if RCV is to be supported or not.
+        ///
+        /// # Errors
+        /// - `UnauthorizedAsAgent` if `origin` is not `ticker`'s sole CAA (owner is not necessarily the CAA).
+        /// - `NoSuchBallot` if `ca_id` does not identify a ballot.
+        /// - `VotingAlreadyStarted` if `start >= now`, where `now` is the current time.
+        #[weight = 950_000_000]
+        pub fn change_rcv(origin, ca_id: CAId, rcv: bool) {
+            // Ensure origin is CAA, a ballot exists, start is in the future.
+            let caa = <CA<T>>::ensure_ca_agent(origin, ca_id.ticker)?;
+            Self::ensure_ballot_not_started(Self::ensure_ballot_exists(ca_id)?)?;
+
+            // Commit to storage + emit event.
+            RCV::insert(ca_id, rcv);
+            Self::deposit_event(Event::<T>::RCVChanged(caa, ca_id, rcv));
+        }
+
+        /// Remove the ballot of the CA identified by `ca_id`.
+        ///
+        /// ## Arguments
+        /// - `origin` which must be a signer for the CAA of `ca_id`.
+        /// - `ca_id` identifies the attached ballot's CA.
+        ///
+        /// # Errors
+        /// - `UnauthorizedAsAgent` if `origin` is not `ticker`'s sole CAA (owner is not necessarily the CAA).
+        /// - `NoSuchBallot` if `ca_id` does not identify a ballot.
+        #[weight = 950_000_000]
+        pub fn remove_ballot(origin, ca_id: CAId) {
+            // Ensure origin is CAA + ballot exists.
+            let caa = <CA<T>>::ensure_ca_agent(origin, ca_id.ticker)?;
+            Self::ensure_ballot_exists(ca_id)?;
+
```

Comment:
We should ensure the start_date is in the future before allowing ballots to be removed.

Reply:
(Will address in a follow-up PR.)